### PR TITLE
HueBridgeHandler: Fix isEqual(state,state) Method

### PR DESCRIPTION
**Pull request by sja**
Before, every Hue light state which was checked for equality resulted in true (=equal) if the light did not support color or alert.
This is not the full truth: Thanks to the `&&` expression, the isEqual() order of equal checking was:

 1. alert
 2. isOn
 3. effect
 4. brightness
 5. color mode
 6. ...

So it only fails if nothing before color mode was not equals.

Signed-off-by: Sebastian Janzen <sebastian@janzen.it>

### CODE REVIEW

**Comment by sjka** [comment]
the jue library really is not nice here... 
lgtm

**Comment by sja** [comment]
@sjka Indeed. And all hue libs stopped developments 2-3 years ago, because Philips opened their API. So usually we habe to develop our own code for that.
The wiki page of Jue also mentions, that it's not up to date with the currenty Philip API. So we'll have to do this step at medium term.

File: extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueBridgeHandler.java [hunk 1]
```diff
@@ -435,18 +435,40 @@ public void startSearch(List<String> serialNumbers) {
         }
     }
 
+    /**
+     * Because the State can produce NPEs on getColorMode() and getAlertMode(), at first we check for the common
```
Comment (maggu2810) [inline extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueBridgeHandler.java]: You have written getAlertMode() could trigger a NPE.
Your implementation seems that you mean getEffect().

Or do I miss something here.

File: extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueBridgeHandler.java [hunk 2]
```diff
@@ -435,18 +435,40 @@ public void startSearch(List<String> serialNumbers) {
...
+     * properties which are set for every light type. If they equal, we additionally try to check the colorMode. If we
+     * get an NPE,
+     * the light does not support color mode and the common properties equality is our result: true. Otherwise if no NPE
+     * occurs
+     * the equality of colorMode is our result.
+     * 
+     * @param state1 Reference state
+     * @param state2 State which is checked for equality.
+     * @return True if both states are equal.
```
Comment (maggu2810) [inline extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueBridgeHandler.java]: if the available informations of both states are equal.

**Comment by sja** [comment]
@maggu2810 Fixed that. Ready for merge?

**Comment by maggu2810** [comment]
All fine

**Pull request merged**